### PR TITLE
a tentative Release howto guide -- a bit long

### PR DESCRIPTION
**Pull request by gasche**
@bobot this would be my reply (too long as usual) to [you wondering](https://github.com/ocaml/ocamlbuild/pull/51#issuecomment-186163361) about a release guide.

### CODE REVIEW

**Comment by gasche** [comment]
(Some part could be extracted to an independent CONTRIBUTING.md document, but let's focus on the need for a release first.)

File: Changes [hunk 1]
```diff
@@ -23,6 +23,8 @@ NEXT_RELEASE:
 OCamlbuild 0.9.0 (19 Jan 2016):
 -------------------------------
 
+- #44: When ocamlbuild discover a cycle in the dependencies, the
```
Comment (hcarty) [inline Changes]: Should be "discovers"

Comment (gasche) [inline Changes]: Fixed, thanks. (Github seems confused about which patches are part of this PR, this one actually isn't, but that's fine)

**Comment by gasche** [comment]
In 9ebb6146f38b455f5a8c1c7bddace49fad71d5b9 I added a release summary. @bobot , if you agree with the steps proposed in the release summary, you should be able to do the rest of the release work. I think it would be a good test.

**Comment by gasche** [comment]
I realize that the howto does not mention that the version number is written in some places of the repo and needs to be updated. `git grep -F "0.9"` is a good way to find those places; currently `META` and `VERSION` need to be updated at each release. I should update the howto.

**Pull request merged**